### PR TITLE
chore: add OTPv2 pojos

### LINKED ISSUE #1: Implement OTP v2 POJOs as Kotlin Multiplatform Mobile
This library should include the following:
* POJOs for OTP REST API v2 in Kotlin
* Client code for managing HTTP request and response and parsing into the POJOs classes. We can use Jackson data binding for this. We should use the Builder pattern.

~~We could pattern this library after our Pelias Client library~~ (**EDIT** - as mentioned below after thinking further we're going to try and make this a Kotlin Multiplatform project, so the below libraries aren't as relevant):
https://github.com/CUTR-at-USF/pelias-client-library

...although ideally I'd like to use Gradle instead of Maven for the build system.

You can see how this library is used in other Java projects here:
https://github.com/CUTR-at-USF/pelias-client-library-demo

This stand-alone project that will be hosted in this repo grew out of initial work to extend our original OTP POJOs project from covering OTP1 REST API to OTP2 REST API:
https://github.com/CUTR-at-USF/opentripplanner-pojos/pull/34

However, the two versions are different enough that IMHO it makes sense to spin it off into a new project that is more than just POJOs - it will include the code to make the HTTP request/response as well, since we need this in multiple projects anyway (https://github.com/CUTR-at-USF/gnss-ride-hailing-analyzer, https://github.com/OneBusAway/onebusaway-android).

This issue can cover the POJOs and we can open a pull request to implement them first, then merge and iterate. I'll open a separate issue for the client code (see https://github.com/CUTR-at-USF/opentripplanner-client-library/issues/2).

### PR DESCRIPTION
**Pull request by BumbleFlash**
This PR adds all the pojos migrated from the opentripplanner-pojos library. 

"closes #1"

### CODE REVIEW

**Comment by BumbleFlash** [comment]
@barbeau What's your opinion on using [retrofit2](https://square.github.io/retrofit/) for making REST API calls? I have used this library in Android and it reduces a lot of lines of code.

**Comment by barbeau** [comment]
Retrofit is fine, although I believe the most modern paradigm now would actually be Kotlin coroutines. I started looking at Kotlin multiplatform as well because there are iOS apps that could also make use of this:
https://www.raywenderlich.com/19144111-kotlin-multiplatform-project-for-android-and-ios-getting-started

See the "Fetching Data From the Network in Common Code" section.

File: src/main/kotlin/edu/usf/cutr/otp/planner/model/core/TraverseMode.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package edu.usf.cutr.otp.planner.model.core
```
Comment (barbeau) [inline src/main/kotlin/edu/usf/cutr/otp/planner/model/core/TraverseMode.kt:1]: Let's just use the current enum values from OTP2:
https://github.com/opentripplanner/OpenTripPlanner/blob/548e209458ac02634330f804e384120fd328500a/src/main/java/org/opentripplanner/routing/core/TraverseMode.java

File: src/main/kotlin/edu/usf/cutr/otp/planner/model/core/OptimzeType.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package edu.usf.cutr.otp.planner.model.core
+
+enum class OptimizeType {
```
Comment (barbeau) [inline src/main/kotlin/edu/usf/cutr/otp/planner/model/core/OptimzeType.kt:3]: There isn't a direct match in OTP2 for this, but I think these enum values should cover most of the responses:
https://github.com/opentripplanner/OpenTripPlanner/blob/4c22825aef84619e1ac3792ede23555ed6f1f91d/src/main/java/org/opentripplanner/routing/core/BicycleOptimizeType.java

**Comment by barbeau** [comment]
Closing this in favor of the Kotlin multiplatform approach, which we're pursuing in https://github.com/CUTR-at-USF/opentripplanner-client-library/pull/4

**Pull request closed without merging**